.github/workflows/build-and-publish-website.yml

```diff
@@ -0,0 +1,108 @@
+      - "website/**"
+      - "firebase.json"
+      - ".github/workflows/*-website.yml"
+
+jobs:
+  website-tests:
+    name: Website Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          target: wasm32-unknown-unknown
+          override: true
+          profile: minimal
+
+      - uses: actions/cache@v2
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target
+          key: cargo-${{ runner.os }}-website-tests-${{ hashFiles('**/Cargo.toml') }}
+          restore-keys: |
+            cargo-${{ runner.os }}-website-tests
+            cargo-${{ runner.os }}-
+
+      - name: Timestamp Cached Target
+        run: ./ci/cargo-sweep sweep -s
+
+      - name: Run website code snippet tests
+        uses: actions-rs/cargo@v1
+        with:
+          command: test
+          args: -p website-test
+
+      - name: Remove Untouched Cache
+        run: ./ci/cargo-sweep sweep -f
+
+  build:
+    needs: [website-tests]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "14.x"
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-build-website-${{ hashFiles('website/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-website-
+            ${{ runner.os }}
+
+      - name: Build
+        run: |
+          cd website
+          npm install
+          npm run build
+
+      - if: github.event.workflow_run.event == 'pull_request'
+        name: Apply pull request environment
+        run: |
+          pr_number=${{ github.event.number }}
+          if ! [[ "$pr_number" =~ ^[0-9]+$ ]]; then
+            echo "pr number invalid"
+            exit 1
+          fi
+          echo "PR_NUMBER=$pr_number" >> $GITHUB_ENV
+          echo "PR_BRANCH=${{ github.event.workflow_run.head_branch }}" >> $GITHUB_ENV
+          echo "COMMIT_SHA=${{ github.event.workflow_run.head_sha }}" >> $GITHUB_ENV
+
+      - if: github.event.workflow_run.event == 'push'
+        name: Apply push environment
+        run: |
+          echo "CHANNEL_ID=live" >> $GITHUB_ENV
+
+      - name: Deploy to Firebase
+        uses: siku2/action-hosting-deploy@v1
+        with:
+          repoToken: "${{ secrets.GITHUB_TOKEN }}"
+          firebaseToken: "${{ secrets.FIREBASE_TOKEN }}"
+          targets: website
+          channelId: "${{ env.CHANNEL_ID }}"
+          # link to the next version because that's what we care about
+          commentURLPath: "/docs/next"
+          # PR information
+          prNumber: "${{ env.PR_NUMBER }}"
+          prBranchName: "${{ env.PR_BRANCH }}"
+          commitSHA: "${{ env.COMMIT_SHA }}"
```

Comment:
The entire reason this was in a separate action is because this won't work when triggered by a PR. Secrets won't be available so the deployment will fail.

Reply:
ahhhhh I was just figuring out why the bot doesn't comment the preview url